dace/dtypes.py

```diff
@@ -815,16 +821,24 @@ def _parse_field_and_types(self, **fields_and_types):
 
     def as_ctypes(self):
         """ Returns the ctypes version of the typeclass. """
+        if self in _FFI_CTYPES:
+            return _FFI_CTYPES[self]
         # Populate the ctype fields for the struct class.
         fields = []
         for k, v in self._data.items():
             if isinstance(v, pointer):
-                fields.append((k, ctypes.c_void_p))  # ctypes.POINTER(_FFI_CTYPES[v.type])))
+                if isinstance(v._typeclass, struct):
+                    fields.append((k, ctypes.POINTER(v._typeclass.as_ctypes())))
+                else:
+                    fields.append((k, ctypes.c_void_p))
+            elif isinstance(v, struct):
+                fields.append((k, v.as_ctypes()))
             else:
                 fields.append((k, _FFI_CTYPES[v.type]))
-        fields = sorted(fields, key=lambda f: f[0])
+        # fields = sorted(fields, key=lambda f: f[0])
         # Create new struct class.
         struct_class = type("NewStructClass", (ctypes.Structure, ), {"_fields_": fields})
+        _FFI_CTYPES[self] = struct_class
```

Comment:
It is not an optimization. Without it the system doesn't work at all because every call to `type` returns a different class (with the same name, inheritance, and fields). Therefore, we need to make it so only one class is generated per Structure type.

Reply:
I see now, so it avoids creating multiple struct_classes. A comment in the code would be great.